task-impls/src/view_sync.rs

```diff
@@ -386,12 +393,18 @@ where
                             ));
 
                             let filter = FilterEvent::default();
-                            let _builder =
+                            let builder =
                                 TaskBuilder::<ViewSyncRelayTaskStateTypes<TYPES, I>>::new(name)
-                                    .register_event_stream(self.event_stream.clone(), filter)
+                                    .register_event_stream(relay_state.event_stream.clone(), filter)
+                                    .await
+                                    .register_registry(&mut self.registry.clone())
                                     .await
                                     .register_state(relay_state)
                                     .register_event_handler(relay_handle_event);
+
+                            // TODO ED For now we will not await these futures, in the future we can await them only in the case of shutdown
+                            let _view_sync_relay_task =
+                                ViewSyncRelayTaskStateTypes::build(builder).launch();
```

Comment:
I think this is where we need to async_spawn and await or the future won't start running

Reply:
yes, you're right!  I'll add that in! Slipped my mind